x-pack/plugins/uptime/server/lib/requests/get_journey_details.ts

```diff
@@ -35,12 +40,18 @@ export const getJourneyDetails: UMElasticsearchQueryFn<
       },
     },
     size: 1,
+    _source: ['synthetics', 'monitor', '@timestamp'],
   };
 
   const { body: thisJourney } = await uptimeEsClient.search({ body: baseParams });
 
-  if (thisJourney?.hits?.hits.length > 0) {
-    const thisJourneySource: any = thisJourney.hits.hits[0]._source;
+  if (thisJourney.hits.hits.length > 0) {
+    const { _id, _source } = thisJourney.hits.hits[0];
+    const decodedJourneySource = JourneyStepType.decode(Object.assign({ _id }, _source));
+    if (!isRight(decodedJourneySource)) {
```

Comment:
We already do this on client side, when calling the rest api, you can pass the io-ts type and client does the validation, i am not sure we need it here.

Reply:
I guess my preference is that it makes it cleaner to simply call these decoders and reference the fields elsewhere with confidence, rather than telling TS to assume they match the right type with something like `return unknownData as MyType`. I can remove these checks and do that though if we feel they're going to be a hinderance.

Reply:
i think too strong of a validation to do on a response. A small change in the data will break api. Imagine if user add some post processor to add some additional data or remove extra data. 

This will end up breaking the API. i would rather let it fail on the client side where we can handle it on UI. 

Also the response run time validation we have on client is also not obstructive. in a sense that it doesn't breaks the api response. it just logs the warning in console.